service/dap/server.go

```diff
@@ -351,13 +361,49 @@ func (s *Server) Run() {
 	}()
 }
 
+func (s *Server) loopHandleRequests(requests <-chan dap.RequestMessage) {
+	for {
+		select {
+		case req, ok := <-requests:
+			if !ok {
+				return
+			}
+			s.handleRequest(req)
+		case <-s.stopTriggered:
+			return
+		}
+	}
+}
+
+func (s *Server) loopHandleResponses(responses <-chan dap.ResponseMessage) {
+	for {
+		select {
+		case res, ok := <-responses:
+			if !ok {
+				return
+			}
+			s.handleResponse(res)
+		case <-s.stopTriggered:
+			return
+		}
+	}
+}
+
 // serveDAPCodec reads and decodes requests from the client
 // until it encounters an error or EOF, when it sends
 // a disconnect signal and returns.
 func (s *Server) serveDAPCodec() {
+	requests := make(chan dap.RequestMessage, 10)
```

Comment:
const maxNumQueuedRequests = 10 (same below)

Reply:
done - or do you prefer them to be defined in the package level?